internal/security/secretstore/certs.go

```diff
@@ -0,0 +1,220 @@
+	client    internal.HttpCaller
+	certPath  string
+	tokenPath string
+}
+
+type auth struct {
+	Token string `json:"root_token"`
+}
+
+func NewCerts(caller internal.HttpCaller, certPath string, tokenPath string) Certs {
+	return Certs{client: caller, certPath: certPath, tokenPath: tokenPath}
+}
+
+func (cs *Certs) certPathUrl() (string, error) {
+	baseURL, err := url2.Parse(Configuration.SecretService.GetSecretSvcBaseURL())
+	if err != nil {
+		e := fmt.Errorf("error parsing secret-service url.  check server and port properties")
+		LoggingClient.Error(e.Error())
+		return "", err
+	}
+
+	certPath, err := url2.Parse(cs.certPath)
+	if err != nil {
+		e := fmt.Errorf("error parsing secret-service certpath.  check certpath property")
+		LoggingClient.Error(e.Error())
+		return "", err
+	}
+
+	fullUrl := baseURL.ResolveReference(certPath)
+	return fullUrl.String(), nil
+}
+
+func (cs *Certs) retrieve(t string) (*CertPair, error) {
+	url, err := cs.certPathUrl()
+	if err != nil {
+		return nil, err
+	}
+
+	req, err := http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		e := fmt.Errorf("error creating http request: %v", err.Error())
+		LoggingClient.Error(e.Error())
+		return nil, e
+	}
+
+	req.Header.Set(VaultToken, t)
+	resp, err := cs.client.Do(req)
+	if err != nil {
+		e := fmt.Errorf("failed to retrieve the proxy cert on path %s with error %s", cs.certPath, err.Error())
+		LoggingClient.Error(e.Error())
+		return nil, e
+	}
+	defer resp.Body.Close()
+
+	cc := CertCollect{}
+
+	if resp.StatusCode == http.StatusNotFound {
+		e := fmt.Errorf("proxy cert pair NOT found in secret store @/%s, status: %s", cs.certPath, resp.Status)
+		LoggingClient.Info(e.Error())
+	} else if resp.StatusCode != http.StatusOK {
+		e := fmt.Errorf("failed to retrieve the proxy cert pair on path %s with error code %d", cs.certPath, resp.StatusCode)
+		LoggingClient.Error(e.Error())
+		return nil, e
+	}
+
+	if err = json.NewDecoder(resp.Body).Decode(&cc); err != nil {
+		e := fmt.Errorf("Error decoding json response when retrieving proxy cert pair: %s", err.Error())
+		LoggingClient.Error(e.Error())
+		return nil, err
+	}
+
+	return &cc.Pair, nil
+}
+
+func (cs *Certs) AlreadyinStore() (bool, error) {
+	cp, err := cs.getCertPair()
+	if err != nil {
+		return false, err
+	}
+	if len(cp.Cert) > 0 && len(cp.Key) > 0 {
+		return true, nil
+	}
+	return false, nil
+}
+
+func (cs *Certs) getCertPair() (*CertPair, error) {
+	t, err := GetAccessToken(cs.tokenPath)
+	if err != nil {
+		return &CertPair{"", ""}, err
+	}
+	cp, err := cs.retrieve(t)
+	if err != nil {
+		return &CertPair{"", ""}, err
+	}
+	return cp, nil
+}
+
+func (cs *Certs) ReadFrom(certPath string, keyPath string) (*CertPair, error) {
+	certPEMBlock, err := ioutil.ReadFile(certPath)
+	if err != nil {
+		return nil, err
+	}
+	cert := string(certPEMBlock)
+
+	keyPEMBlock, err := ioutil.ReadFile(keyPath)
+	if err != nil {
+		return nil, err
+	}
+	key := string(keyPEMBlock)
+
+	cp := CertPair{
```

Comment:
minor nitpick, you could just include this in the return statement rather than define a new variable to return